identity/src/test/java/com/android/identity/UtilTest.java

```diff
@@ -783,6 +783,49 @@ public void fromHexThrows() {
         assertThrows(IllegalArgumentException.class, () -> Util.fromHex("XX"));
     }
 
+    @Test
+    public void asString() {
+        final DataItem dataItem = new co.nstant.in.cbor.model.UnicodeString("hello");
+        assertEquals( "hello", Util.asString(dataItem));
+    }
+
+    @Test
+    public void asStringThrows() {
+        assertThrows(IllegalArgumentException.class,
+            () -> Util.asString(co.nstant.in.cbor.model.SimpleValue.TRUE));
+        assertThrows(IllegalArgumentException.class,
+            () -> Util.asString(new co.nstant.in.cbor.model.UnsignedInteger(42)));
+        assertThrows(IllegalArgumentException.class,
+            () -> Util.asString(new co.nstant.in.cbor.model.ByteString(new byte[] { 1, 2, 3 })));
+    }
+
+    @Test
+    public void asLong() {
+        final DataItem dataItem = new co.nstant.in.cbor.model.UnsignedInteger(8675309);
```

Comment:
Is UnsignedInteger the right class? I see it gets constructed from an int, but returns a BigInteger from getValue(). Can we have a DataItem that wraps a BigInteger and try the boundary cases, i.e. 

BigInteger.valueOf(Long.MIN_VALUE)
BigInteger.valueOf(Long.MAX_VALUE)
BigInteger.valueOf(Long.MIN_VALUE).subtract(BigInteger.ONE)
BigInteger.valueOf(Long.MAX_VALUE).add(BigInteger.ONE)

?

Reply:
Hah, I just saw this comment after I coded this exact thing :) Done!

Note that CBOR types negative and unsigned integers separately. `Number` is the base for all numbers, but it's abstract (and could never be concrete, I think).

I'm not certain why, but I think CBOR does this to aggressively optimize every single bit. By dropping the sign bit, we can encode larger integers in less space.